### ISSUE TITLE
ES6 Shim breaks build. Safari is not working without it

### ISSUE THREAD

**Issue by talmog** [antonybudianto/angular2-starter#40]
## Issue Overview
My project is not working on safari. It complains about the Router.
When I include es6-shim `<script src="node_modules/es6-shim/es6-shim.js"></script>` in index.html as recommended by angular (https://angular.io/docs/ts/latest/quickstart.html), and run `gulp` everything works fine on all browsers

When I run `gulp build`, everything breaks, and I get errors also on chrome.

## Issue Description
```
Errors on safari without es6-shim:
[Error] EXCEPTION: Error during instantiation of Router! (RouterOutlet -> Router).
[Error] ORIGINAL EXCEPTION: TypeError: undefined is not a function (evaluating 'route.startsWith("/")')
```
...

Errors on Chrome when es6-shim is included:
```
Uncaught TypeError: Cannot read property 'apply' of undefinedIr @ bundle.js:2(anonymous function) @ bundle.js:2(anonymous function) @ bundle.js:1(anonymous function) @ bundle.js:2(anonymous function) @ bundle.js:1(anonymous function) @ bundle.js:1
(index):19 Uncaught ReferenceError: System is not defined
```

<Please explain the way to reproduce your issue, or even better with Plunker link>

## Information
|                      	|                                 	|
|---------------------	|---------------------------------	|
| **Operating System** 	| OSX 	|
| **Node version**     	| 5.2.0	|
| **NPM Version**      	| 3.5.3  	|
| **Environment**       | Browser  |

## THANK YOU!

**Comment by antonybudianto** [comment]
Thank you for reporting this issue!
I believe this is related to angular2 [issue](https://github.com/angular/angular/issues/6706)
I will keep this open to track it

**Comment by antonybudianto** [comment]
Also, may I know which line number you put the es6 shim in `index.html`?

**Comment by antonybudianto** [comment]
if you put es6-shim script reference below systemjs, it works

**Comment by talmog** [comment]
That didn't work for me, but putting it before everything abive `<!-- build:js assets/bundle.js -->` did!
Thank you very much!
one more quick questions: how come you have `<!-- build:remove -->` around rx and angular? where will angular load from if you remove it at build?

**Comment by antonybudianto** [comment]
I use systemjs builder to bundle them, refer [here](https://github.com/antonybudianto/angular2-starter/blob/master/gulp.config.js#L70), that's why I remove them when bundling (`gulp build` or `gulp serve-build`) since it's only used when development

**Comment by talmog** [comment]
I see, so if I use angular router or http should the be in the bundle or removed on build?

**Comment by talmog** [comment]
Thank you!

**Comment by antonybudianto** [comment]
my pleasure :)
just FYI, if you happen to need another angular2 [bundles](https://github.com/angular/angular/blob/master/modules/angular2/docs/bundles/overview.md#modules-barrels-and-bundles) like `angular2/http`, just put the script tag inside `<!-- build:remove -->`, the bundling process will be handled by wildcard so don't need to put it in the config again

**Comment by talmog** [comment]
Awesome, thank you so much!

**Comment by rdai** [comment]
I don't know if it is ok to comment on a closed issue or better starting a new one. 

I am very new to many of the libraries and features, when I did gulp build of the vanilla git clone and copy the build folder to my remote apache server, the index.html won't load with the error

"index.html:20 Uncaught ReferenceError: System is not defined(anonymous function) @ index.html:20"

It works locally with "gulp servedev", since it looks similar to this issue, I tried to move the es6-shim around but it is still not working. 

Any ideas? Thanks!
